### ISSUE TITLE
[Bug]: Null ls stays active even on files it wasnt originally active on if it was activated by a different file

### ISSUE THREAD

**Issue by afridley** [LunarVim/LunarVim#1309]
### Problem description

Add linters and prettier to null-ls in config
```lua
lvim.lang.javascript.linters = {
  {
    exe = "eslint", -- can be eslint_d as well
    args = {},
  },
}
lvim.lang.javascriptreact.linters = {
  {
    exe = "eslint", -- can be eslint_d as well
    args = {},
  },
}
lvim.lang.typescript.linters = {
  {
    exe = "eslint", -- can be eslint_d as well
    args = {},
  },
}
lvim.lang.typescriptreact.linters = {
  {
    exe = "eslint", -- can be eslint_d as well
    args = {},
  },
}
lvim.lang.javascript.formatters = {
  {
    exe = "prettier", -- can be prettierd eslint, or eslint_d as well
    args = {},
  },
}
lvim.lang.javascriptreact.formatters = {
  {
    exe = "prettier", -- can be prettierd eslint, or eslint_d as well
    args = {},
  },
}

lvim.lang.typescript.formatters = {
  {
    exe = "prettier", -- can be prettierd, eslint or eslint_d as well
    args = {},
  },
}
lvim.lang.typescriptreact.formatters = {
  {
    exe = "prettier", -- can be prettierd, eslint or eslint_d as well
    args = {},
  },
}
```

Now if you have say a .svelte file that should not be using eslint or prettier mixed in a project with ts files you will get the following behavior

open the svelte file everything is good and functioning correctly

open a ts file and it will start null-ls
But if you go back to the svelte file eslint and prettier are still active in null-ls and you start to see false errors. saving will also ask which language server you would wish to use svelte or null-ls

This also happens in scss files where null-ls just crashes trying to load prettier and cssls 
```error executing vim.schedule lua callback: /home/user/.local/share/lunarvim/lvim/lua/lsp/null-ls.lua:95 attempt to index a nil value```

tldr null-ls does not start if you open a .svelte file or .scss file but once you do open a file that starts null-ls it starts having problems

Null-ls should be able to function alongside other lang servers providing the correct diagnostics and code actions similar to what efm did in the past

Not sure if this is a bug with null ls or our config

### LunarVim version

5.1 rolling 8/12/2020

### Neovim version (>= 0.5)

.5

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

**Comment by kylo252** [comment]
This is a tricky one to solve, because by default these providers have a list of supported filetypes that are not really configured by LunarVim. See: 

https://github.com/jose-elias-alvarez/null-ls.nvim/blob/411658dc8a96a70cc47f16bc21f8634fdd4157fd/lua/null-ls/builtins/formatting.lua#L347-L362

I'm not sure this use-case is worth the added overhead of keeping track of which file to support or not, and if we make it so that it only works on the requested filetype then you have people on the other side of the fence who wouldn't want to spawn a new prettier instance per filetype. 

So you might be better off either defining a custom provider, or setting up null-ls manually for the time being.

**Comment by Tastyep** [comment]
Is this still true after my refactor ?
Since [here](https://github.com/LunarVim/LunarVim/blob/rolling/lua/lsp/null-ls/formatters.lua#L70) I overwrite the formatters (and linters in the other file), I would expect that only the configured provider would be selected.

**Comment by afridley** [comment]
@Tastyep This is now fixed thank you sir :)
